ui/packages/atlasmap/src/UI/dialogs/ParametersDialog.tsx

```diff
@@ -0,0 +1,178 @@
+import React, {
+  FunctionComponent,
+  useState,
+  useCallback, useEffect,
+} from "react";
+
+import {
+  Button,
+  Form,
+  FormGroup, FormSelect, FormSelectOption, Split, SplitItem,
+  TextInput,
+} from "@patternfly/react-core";
+
+import {
+  ConfirmationDialog,
+  IConfirmationDialogProps,
+} from "./ConfirmationDialog";
+import {PlusIcon, TimesIcon} from "@patternfly/react-icons";
+
+export interface IParameter {
+  name: string;
+  value: string;
+  options?: IParameterOption[];
+  hidden?: boolean;
+  hideable?: boolean;
+}
+
+export interface IParameterOption {
+  label: string;
+  value: string;
+}
+
+export interface IParametersDialogProps {
+  title: string;
+  parameters?: IParameter[];
+  isOpen: IConfirmationDialogProps["isOpen"];
+  onCancel: IConfirmationDialogProps["onCancel"];
+  onConfirm: (parameters: IParameter[]) => void;
+}
+export const ParametersDialog: FunctionComponent<IParametersDialogProps> = ({
+  title,
+  parameters: initialParameters = [],
+  isOpen,
+  onCancel,
+  onConfirm,
+}) => {
+  const initialAddParameter = 'Select...';
+
+  const [parameters, setParameters] = useState(initialParameters);
+
+  const [addParameter, setAddParameter] = useState(initialAddParameter);
+
+  const reset = useCallback(() => {
+    setParameters(initialParameters);
+  }, [initialParameters]);
+
+  const handleOnConfirm = useCallback(() => {
+    onConfirm(parameters);
+  }, [parameters, onConfirm]);
+
+  const handleOnCancel = useCallback(() => {
+    onCancel();
+    reset();
+  }, [onCancel, reset]);
+
+  const handleAddParameter = useCallback(() => {
+    const newParameters = [...parameters];
+    for (const newParameter of newParameters) {
+      if (newParameter.name === addParameter) {
+        newParameter.hidden = false;
+        break;
+      }
+    }
+    setAddParameter(initialAddParameter);
+    setParameters(newParameters);
+  }, [parameters, addParameter]);
+
+  const handleRemoveParameter = useCallback((name) => {
```

Comment:
Same as above. Unless I'm missing something, that `hidden` property with the proposed changes is not needed anymore.